mm/2s2h/Rando/ActorBehavior/ObjTree.cpp

```diff
@@ -0,0 +1,241 @@
+    { { SCENE_17SETUGEN2, 0, 8 }, RC_TWIN_ISLANDS_SPRING_TREE_02 },
+    { { SCENE_17SETUGEN2, 0, 9 }, RC_TWIN_ISLANDS_SPRING_TREE_03 },
+    { { SCENE_17SETUGEN, 0, 7 }, RC_TWIN_ISLANDS_TREE_01 },
+    { { SCENE_17SETUGEN, 0, 8 }, RC_TWIN_ISLANDS_TREE_02 },
+    { { SCENE_17SETUGEN, 0, 9 }, RC_TWIN_ISLANDS_TREE_03 },
+
+    // Zora Cape
+    { { SCENE_31MISAKI, 0, 24 }, RC_ZORA_CAPE_TREE_01 },
+    { { SCENE_31MISAKI, 0, 25 }, RC_ZORA_CAPE_TREE_02 },
+    { { SCENE_31MISAKI, 0, 26 }, RC_ZORA_CAPE_TREE_03 },
+    { { SCENE_31MISAKI, 0, 27 }, RC_ZORA_CAPE_TREE_04 },
+    { { SCENE_31MISAKI, 0, 28 }, RC_ZORA_CAPE_TREE_05 },
+};
+
+std::map<RandoCheckId, EnItem00*> currentlySpawnedItems;
+
+void ApplyGravityToSpawnedItem(RandoCheckId randoCheckId) {
+    auto findItem = currentlySpawnedItems.find(randoCheckId);
+    if (findItem != currentlySpawnedItems.end()) {
+        findItem->second->actor.gravity = -1.4f;
+    }
+}
+
+void SpawnTreeDrop(Vec3f pos, RandoCheckId randoCheckId) {
+    int customItemFlags = CustomItem::KILL_ON_TOUCH | CustomItem::ABLE_TO_ZORA_RANG;
+
+    if (RANDO_SAVE_OPTIONS[RO_PEEK_TREE_DROPS] == RO_GENERIC_OFF) {
+        customItemFlags |= CustomItem::TOSS_ON_SPAWN;
+    }
+
+    EnItem00* spawnedItem = CustomItem::Spawn(
+        pos.x - 20.0f, pos.y + 200.0f, pos.z + 25.0f, 0, customItemFlags, randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+
+    auto findItem = currentlySpawnedItems.find(randoCheckId);
+    if (findItem == currentlySpawnedItems.end()) {
+        currentlySpawnedItems.insert({ randoCheckId, spawnedItem });
+    }
+
+    if (randoCheckId == RC_TERMINA_FIELD_MAN_IN_THE_TREE_RUPEE_01) {
+        pos.x += 40.0f;
+        SpawnTreeDrop(pos, RC_TERMINA_FIELD_MAN_IN_THE_TREE_RUPEE_02);
+    }
+}
+
+void IdentifyTree(Actor* actor, bool* should) {
+    s16 actorListIndex = GetActorListIndex(actor);
+    RandoCheckId randoCheckId = RC_UNKNOWN;
+
+    auto it = treeActorIdMap.find({ gPlayState->sceneId, gPlayState->roomCtx.curRoom.num, actorListIndex });
+    if (it != treeActorIdMap.end()) {
+        randoCheckId = it->second;
+    }
+
+    if (!RANDO_SAVE_CHECKS[randoCheckId].shuffled || RANDO_SAVE_CHECKS[randoCheckId].cycleObtained) {
+        return;
+    }
+
+    if (!RANDO_SAVE_CHECKS[randoCheckId].obtained && RANDO_SAVE_OPTIONS[RO_PEEK_TREE_DROPS] == RO_GENERIC_ON) {
+        SpawnTreeDrop(actor->world.pos, randoCheckId);
+    }
+
+    Rando::ActorBehavior::GetObjectRandoCheckId(actor);
```

Comment:
Should this be SetObjectRandoCheckId? How does VB_APPLY_BONK_TO_ACTOR or VB_TREE_DROP_COLLECTIBLE know which tree was bonked?
```suggestion
    Rando::ActorBehavior::SetObjectRandoCheckId(actor, randoCheckId);
```

Reply:
Yea this is the right usage. the actor comes from the VB in z_player. its called cylinderOc (basically just the actor lol)